### ISSUE TITLE
Consideration and explanation of: failed validation causes exception that is logged as error by LoggingBehaviour

### ISSUE THREAD

**Issue by NikiforovAll** [jasontaylordev/CleanArchitecture#324]
**Is your feature request related to a problem? Please describe.**
When I check logs, I see errors and long error stack traces for validation failed scenarios. I'm a little bit frustrated because it is a common and expected scenario for web applications.

**Describe the solution you'd like**
I would like to know what are recommendations and further/possible steps to improve logging.

**Describe alternatives you've considered**
One option I consider is to enable validation via `[ApiController]` and `SuppressModelStateInvalidFilter = false;`. But this is strange because this way I have two places when FluentValidation can be triggered and validation is not part of the application pipeline.

**Additional context**

![image](https://user-images.githubusercontent.com/8037439/111005009-b203f600-8392-11eb-847f-a4c19fd38de9.png)

**Comment by mweel1** [comment]
Would it be helpful to return a result from the controllers?  Once this is in place the model validation middleware could also take advantage?

    public class Result<T>
    {
        public bool IsSuccess { get; private set; }
        public T Value { get; private set; }
        public string[] Errors { get; private set; }

        private Result() { }

        public static Result<T> Success(T value) => new Result<T>() { IsSuccess = true, Value = value };
        public static Result<T> Failure(params string[] errors) => new Result<T>() { IsSuccess = false, Errors = errors };
    }

**Comment by jasontaylordev** [comment]
In this case, validation occurs on both the client and the server. Client validation is managed by Angular, and server validation managed using FluentValidation and the validation behaviour. If validation fails on the client, this is normal and expected - users make mistakes. However, if validation occurs on the server this is unexpected, the client should have validated the database before sending it to the server - there may be a bug or someone has bypassed validation. Either way, this is an exceptional circumstance and this is why the exception is thrown and logged.

If validation failures on the server were normal, then validation exceptions would not be thrown. Instead a result class such as the one shown above would be a good approach.

I hope this helps to explain the reason for this design, feel free to customise it to suit your requirements. Happy coding!
